### PR TITLE
Misc changes

### PR DESCRIPTION
**Pull request by TobsA13**
Hi, 

I have fixed some bugs which occurred during my first use of the software.

Please check the different commits for details.

### CODE REVIEW

File: dali_mqtt_daemon.py [hunk 1]
```diff
@@ -5,6 +5,7 @@
 import io
 import logging
 import re
+
```
Comment (dgomes) [inline dali_mqtt_daemon.py:8]: no need to split

File: dali_mqtt_daemon.py [hunk 2]
```diff
@@ -72,16 +73,19 @@
 logging.basicConfig(format=LOG_FORMAT)
 logger = logging.getLogger(__name__)
 
-def dali_scan(driver):
+def dali_scan(driver, max_lamps):
     """Scan a maximum number of dali devices."""
     lamps = []
-    for lamp in range(0, 63):
+    for lamp in range(0, 64):
```
Comment (dgomes) [inline dali_mqtt_daemon.py:79]: why?

File: consts.py [hunk 3]
```diff
@@ -27,9 +27,12 @@
 DEFAULT_DEVICES_NAMES_FILE = "devices.yaml"
 DEFAULT_MQTT_SERVER = "localhost"
 DEFAULT_MQTT_PORT = "1883"
+DEFAULT_MQTT_USERNAME = ""
+DEFAULT_MQTT_PASSWORD = ""
 DEFAULT_MQTT_BASE_TOPIC = "dali2mqtt"
-DEFAULT_HA_DISCOVERY_PREFIX = "homeassistant"
 DEFAULT_DALI_DRIVER = "hasseb"
+DEFAULT_DALI_LAMPS = 64
+DEFAULT_HA_DISCOVERY_PREFIX = "homeassistant"
```
[the remark this answers is not in the document]
Comment (TobsA13) [reply consts.py:35]: The `DEFAULT_HA_DISCOVERY_PREFIX` line has been moved down so the `CONF_` block and the `DEFAULT_` block have the same order.

File: config.py [hunk 4]
```diff
@@ -39,10 +42,16 @@
         vol.Optional(CONF_MQTT_PORT, default=DEFAULT_MQTT_PORT): vol.All(
             vol.Coerce(int), vol.Range(min=1, max=65535)
         ),
+        vol.Required(CONF_MQTT_USERNAME, default=DEFAULT_MQTT_USERNAME): str,
```
[the remark this answers is not in the document]
Comment (TobsA13) [reply config.py]: Fixed

File: dali_mqtt_daemon.py (hunk 1 above)
Comment (TobsA13) [reply dali_mqtt_daemon.py]: Fixed

File: dali_mqtt_daemon.py [hunk 5]
```diff
@@ -72,16 +73,19 @@
...
         try:
-            logging.debug("Search for Lamp %s", lamp)
+            logger.debug("Search for Lamp %s", lamp)
             present = driver.send(gear.QueryControlGearPresent(address.Short(lamp)))
             if isinstance(present, YesNoResponse) and present.value:
                 lamps.append(lamp)
                 logger.debug("Found lamp at address %d", lamp)
+                if len(lamps) >= max_lamps:
+                    logger.info("Found configured amounts of lamps.")
```
[the remark this answers is not in the document]
Comment (TobsA13) [reply dali_mqtt_daemon.py]: Edit: Made message more clear in the last commit.

File: consts.py [hunk 6]
```diff
@@ -27,9 +27,12 @@
 DEFAULT_DEVICES_NAMES_FILE = "devices.yaml"
 DEFAULT_MQTT_SERVER = "localhost"
 DEFAULT_MQTT_PORT = "1883"
+DEFAULT_MQTT_USERNAME = ""
+DEFAULT_MQTT_PASSWORD = ""
```
[the remark this answers is not in the document]
Comment (dgomes) [reply consts.py:31]: You can adjust the type hinting to `str | None`

of course the code needs to be adjusted not to write down variables that are `None`.

File: dali_mqtt_daemon.py [hunk 7]
```diff
@@ -72,16 +72,19 @@
...
+    for lamp in range(0, 64):
         try:
-            logging.debug("Search for Lamp %s", lamp)
+            logger.debug("Search for Lamp %s", lamp)
             present = driver.send(gear.QueryControlGearPresent(address.Short(lamp)))
             if isinstance(present, YesNoResponse) and present.value:
                 lamps.append(lamp)
                 logger.debug("Found lamp at address %d", lamp)
+                if len(lamps) >= max_lamps:
+                    logger.info("Stopping scan. Found configured amounts of lamps.")
```
Comment (dgomes) [inline dali_mqtt_daemon.py:86]:
```suggestion
                    logger.warning("All %s configured lamps have been found, Stopping scan",  max_lamps)
```

File: consts.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (TobsA13) [reply consts.py:31]: `str | None` does not work. I don't know the voluptuous lib, maybe there is an option to fix this.

But from my point of view, I don't see a reason to program the default in the code to a None if we save still save it as an empty string. And for this add code which converts this during saving and loading.
The empty string for no access credentials is common for many programs.  


```
  File "/root/dali2mqtt/config.py", line 45, in <module>
    vol.Optional(CONF_MQTT_USERNAME, default=DEFAULT_MQTT_USERNAME): str | None,
TypeError: unsupported operand type(s) for |: 'type' and 'NoneType'
```

[the remark this answers is not in the document]
Comment (dgomes) [reply consts.py:31]: Guessing you require https://docs.python.org/3/library/typing.html#typing.Optional

**Comment by dgomes** [comment]
:(

**Pull request closed without merging**